src/modules/user/user.service.ts

```diff
@@ -252,4 +256,56 @@ export class UserService {
       throw new InternalServerErrorException();
     }
   }
+
+  async getFheelancerInformation(
+    take: number,
+    page: number,
+  ): Promise<[User[], number]> {
+    try {
+      const currentUser = await this.userRepository
+        .createQueryBuilder('user')
+        .leftJoinAndSelect('user.category', 'category')
+        .skip((page - 1) * take)
+        .take(take)
+        .getManyAndCount();
+
+      return currentUser;
+    } catch (error) {
+      if (error instanceof HttpException) {
+        throw error;
+      }
+      throw new InternalServerErrorException();
+    }
+  }
+
+  async addCategoryInfo(payload: Category, user: User): Promise<void> {
+    try {
+      const currentUser = await this.findByEmail(user.email);
```

Comment:
no, you already have a user. its too expensive to get a user for each endpoint, find another way. this function calls for the protected route, then you get a user from guard validation, you can use the same user if needed

Reply:
fixed